sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DocumentCollection.java

```diff
@@ -27,6 +29,16 @@
     private UniqueKeyPolicy uniqueKeyPolicy;
     private PartitionKeyDefinition partitionKeyDefinition;
 
+    /**
+     * Constructor.
+     *
+     * @param objectNode the {@link ObjectNode} that represent the
+     * {@link JsonSerializable}
+     */
+    public DocumentCollection(ObjectNode objectNode) {
```

Comment:
Same, and everywhere else. azure-core will not allow exposing `ObjectNode`

Reply:
all of the new public constructors are in the implementation package not in the public surface area.